src/Sentry/Internal/Hub.cs

```diff
@@ -158,6 +158,7 @@ public void WithScope(Action<Scope> scopeCallback)
 
         public SentryTraceHeader? GetSentryTrace()
         {
+            // TODO:
             var (currentScope, _) = ScopeManager.GetCurrent();
             return currentScope.Transaction?.GetTraceHeader();
         }
```

Comment:
@bruno-garcia this previously relied that transaction is added to the current scope.
How should we get current transaction now? Or is it current span? Can you please explain how `Hub.GetSentryTrace()` should work?

Reply:
I believe it needs to read the current span. In a world where we don't bind a whole transaction to the scope (also because it isnt' thread safe) we would only have a tree (in a list form) of spans. We could build this SentryTrace by getting the current span and with that the traceid and sampling decision. Isn't this spec'ed out? How is it in Java and Python?

/cc @marandaneto, @maciejwalkowiak, @untitaker

Reply:
Python assembles the entire trace tree all the time. Regardless of sampling decision it propagates a trace header with outbound HTTP requests that is derived from the current span.

If you feel strongly about performance and don't want to create span IDs when the transaction sampling decision is 0, I think you could take something other than the current span as long as the trace ID is propagated properly.

>In a world where we don't bind a whole transaction to the scope (also because it isnt' thread safe)

This already deviates from Python behavior so fundamentally that I cannot give any advice here. Python scopes are basically threadsafe and the current span is always bound to the scope. This is useful because it means outgoing error events can grab the current span from the scope to attach a trace context that makes sense.

Reply:
Yeah this "Transaction on the Scope" Seem to have originated on JS, then the Spec and now Java and .NET.

The topic wasn't about avoiding allocations in this case but really that, the transaction on the scope (as opposed to only spans).

Reply:
@bruno-garcia all of that makes sense, I was more interested in the `current span` part. Considering it doesn't store the transaction anywhere when you call `StartTransaction`, where would it get the current span?

Reply:
If you want to make the transaction globally accessibly you must bind the transaction (or ideally the span list in a thread-ssafe manner instead, since we will have huge headaches with the Transaction thing on the scope!) to the Scope. It's optional. If it's not bound, it won't be available for integrations unless you programatically pass the value it's needed.

var tran = hub.StartTransction(..);

DoSomething("arg", tran);

void DoSomething(string arg, ITransaction tran)
{
 var client = new HttpClient(new SentryHandler(tran));
 client.Get()..
 // or
  var clietn = new HttpClient();
  var req = new HttpRequest();
  req.Headers.Add(tran.GetSentryTrace());
  client.Request(req);
}

Reply:
Ok I see. So the code is right, it's just that it will rely on the user to set the transaction themselves.